portfolio/src/main/webapp/script.js

```diff
@@ -1,32 +1,265 @@
 // Author: Jonathan Chavez - tabaresj@google.com
 var isTerminalOpen = false;
+var firstTimeOpening = true;
+
+// Names for the sites in the terminal, each is a section of the portfolio
+var sites = [
+  'education.yaml',
+  'work-experience.yaml',
+  'projects.yaml',
+  'skills.yaml',
+  'dangerous-virus.exe',
+];
+
+// List of commands available to use in the terminal
+var commands = ['man', 'compgen', 'ls', 'open', 'whoami'];
 
 window.addEventListener('keydown', keyboardHandler);
 
+/*
+ * Handles keyboard strokes for detecting when toggling the terminal
+ */
 function keyboardHandler(e) {
   e = e || window.event;
-  if (String.fromCharCode(e.keyCode) === 'T') {
+  if (String.fromCharCode(e.keyCode) === 'T' && e.shiftKey) {
     if (!isTerminalOpen) {
       openTerminal();
       isTerminalOpen = true;
-      runTerminal();
+      if (firstTimeOpening) {
+        firstTimeOpening = false;
+        runTerminalTutorial();
+      }
     } else {
       closeTerminal();
       isTerminalOpen = false;
     }
   }
 }
 
-function runTerminal() {}
+/*
+ * Shows a tutorial on the terminal
+ */
+function runTerminalTutorial() {
+  printTerminal(
+    "^500Hola! ^400 I'm Jonathan,^300 nice to meet you.",
+    function () {
+      printTerminal(
+        'This is a terminal, ^500 well^300.^100.^100.^400 kind of.',
+        function () {
+          printTerminal(
+            "To list all commands you can type 'compgen'.",
+            function () {
+              printTerminal(
+                "To learn more about a command type 'man [command]',^500 e.g.:^400 'man compgen'.",
+                function () {
+                  addTerminalPrompt();
+                }
+              );
+            }
+          );
+        }
+      );
+    }
+  );
+}
```

Comment:
I have a question here. This is a function that receives a callback and as you can see I nested a lot of those here although it won't grow more. Do you think I should refactor it?

Reply:
Hii Jonathan,

Well, you've found a pattern jaja, this is usual, and it's called 'Callback hell'. Actually you can google it and see that there are some memes about it.

What you should do for this cases is to convert your printTerminal into a Promise, so you can use async/await or .then syntax.

I can help you with that. I'll send you a message via meet chat.